src/parquet/column/levels.h

```diff
@@ -95,23 +95,32 @@ class LevelDecoder {
  public:
   LevelDecoder() {}
 
-  // Initialize the LevelDecoder and return the number of bytes consumed
-  size_t Init(parquet::Encoding::type encoding, int16_t max_level,
+  // Initialize the LevelDecoder
+  void Init(int16_t max_level) {
+    bit_width_ = BitUtil::Log2(max_level + 1);
+    rle_decoder_.reset(new RleDecoder(NULL, 0, 0));
+    bit_packed_decoder_.reset(new BitReader(NULL, 0));
```

Comment:
Perhaps only initialize the appropriate decoder, and only if `max_level > 0` -- does the Parquet spec indicate whether data pages could switch encodings mid-stream -- in DataPageV2 there is no bit-packed encoding anymore so this is less of a concern there